pages/api/[_id].js

```diff
@@ -0,0 +1,21 @@
+import dbConnect from "../../db/dbConnect";
+import Surfspots from "../../db/Models/Surfspots";
+
+export default async function handler(req, res) {
+  await dbConnect();
+
+  if (req.method === "PUT") {
+    const id = req.query._id;
+    const updatedDocument = await Surfspots.findByIdAndUpdate(
+      id,
```

Comment:
This can fail! You need to wrap your code inside a `try ... catch` block and return an appropriate error code (e.g. 500) in case of unexpected errors (e.g. DB down etc.)